File: solved.ac/Class1/1157.c
// 1157. 단어 공부 (https://www.acmicpc.net/problem/1157)
#include <stdio.h>
#include <string.h>

int main()
{
  int max, result = 0, len;
  char arr[1000000];
  int alp_cnt[26] = {0, };
  int select = 0;

  scanf("%s", arr);
  len = strlen(arr);

  for (int i = 'a'; i<='z';i++){
    for (int j=0;j<len;j++){
      if (i == arr[j]) alp_cnt[i - 'a']++;
    }
  }

  for (int i='A';i<='Z';i++){
    for (int j=0;j<len;j++){
      if (i == arr[j]) alp_cnt[i - 'A']++;
    }
  }

  max = alp_cnt[0];
  for (int i=1;i<26;i++){
    if (max < alp_cnt[i]){
      max = alp_cnt[i];
      select = i;
    }
  }

  for (int i=1;i<26;i++){
    if (max == alp_cnt[i]){
      result++;
    }
  }

  if (result > 1) printf("\n");
  else printf("%c", select +'A');

  return 0;
}

File: solved.ac/Class1/1157.py
## 1157. 단어 공부 (https://www.acmicpc.net/problem/1157)
'''
word = input()
word = word.upper()
array = [0] * 26

for w in word:
  array[ord(w) - 65] += 1

max = 0
max_idx = 0
for idx in range(len(array)):
  if array[idx] > max:
    max = array[idx]
    max_idx = idx
  elif array[idx] == max and max != 0:
    print("?")
    break

  if idx == len(array) - 1:
    print(chr(max_idx + 65))
'''

word = input().upper()
unique_word = list(set(word))

cnt_list = []
for x in unique_word:
  cnt = word.count(x)
  cnt_list.append(cnt)

if cnt_list.count(max(cnt_list)) > 1:
  print("?")
else:
  max_index = cnt_list.index(max(cnt_list))
  print(unique_word[max_index])